src/main/java/emissary/core/sentinel/rules/Rule.java

```diff
@@ -0,0 +1,90 @@
+package emissary.core.sentinel.rules;
+
+import emissary.core.NamespaceException;
+import emissary.core.sentinel.Sentinel;
+import emissary.pool.AgentPool;
+
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.lang.invoke.MethodHandles;
+import java.util.Comparator;
+import java.util.Map;
+import java.util.StringJoiner;
+
+public abstract class Rule {
+
+    protected static final Logger logger = LoggerFactory.getLogger(MethodHandles.lookup().lookupClass());
+
+    private final String place;
+
+    // how long to wait before alerting on stuck agents
+    private final long timeLimit;
+
+    // percentage of mobile agents that are stuck on the same place before sounding the alarm
+    private final double threshold;
+
+    public Rule(String place, long timeLimit, double threshold) {
+        this.place = place;
+        this.timeLimit = timeLimit;
+        this.threshold = threshold;
+        logger.trace("Loaded {}", this);
```

Comment:
I think it would be useful to know which rules were loaded.
```suggestion
        logger.info("Loaded {}", this);
```